### PR TITLE
SONARIAC-1230 Align AzureResourceManager property keys and deprecate old key format

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

File: its/plugin/src/test/java/org/sonarsource/iac/PropertiesTest.java [hunk 1]
```diff
@@ -105,6 +105,31 @@ void testArmSuffix(String projectKey, String language, String suffixes, int expe
     executeBuildAndAssertMetric(projectKey, language, "suffixes", suffixes, "ncloc", expectedNcloc);
   }
 
+  // The active property will always win over the deprecated one if set
+  @ParameterizedTest
+  @CsvSource(delimiter = ';', value = {
+    "armDeprecatedPropertyTestBothDisabled; false; false;",
```
Comment (petertrr) [inline its/plugin/src/test/java/org/sonarsource/iac/PropertiesTest.java:111]: I wonder if we need so many integration tests for this. Is it possible to write a test where only configuration is loaded, or are these details (which property takes precedence) not easily accessible from plugin API?

**Review by petertrr — APPROVED** [review]
LGTM, but also please see my question

File: its/plugin/src/test/java/org/sonarsource/iac/PropertiesTest.java (hunk 1 above)
Comment (jonas-wielage-sonarsource) [reply its/plugin/src/test/java/org/sonarsource/iac/PropertiesTest.java:111]: I don't think we need that many that are executed everytime. I used this also to make sure that they work as I expect.

Comment (jonas-wielage-sonarsource) [reply its/plugin/src/test/java/org/sonarsource/iac/PropertiesTest.java:111]: But they are really cheap to execute as the Orchestrator is already running

Comment (jonas-wielage-sonarsource) [reply its/plugin/src/test/java/org/sonarsource/iac/PropertiesTest.java:111]: I will remove some of them

**Pull request merged**